File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        maxProfit = 0
        
        while r < len(prices): 
            if (prices[r] - prices[l]) > maxProfit:
                maxProfit = prices[r] - prices[l]
            
            if prices[l] > prices[r]:
                l = r
            
            r += 1
            
        return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = 1;
        int maxP = 0;

        while (right < prices.length) {
            if ((prices[right] - prices[left]) > maxP) {
                maxP = prices[right] - prices[left];
            }
            if (prices[right] < prices[left]) {
                left = right;
            }
            right++;
        }

        return maxP;
    }
}
